press/press/doctype/server/server.py

```diff
@@ -811,9 +811,16 @@ def auto_scale_workers(self):
 			self._auto_scale_workers_old()
 
 	def _auto_scale_workers_new(self):
-		usable_ram = max(
-			self.ram - 3000, self.ram * 0.75
-		)  # in MB (leaving some for disk cache + others)
+		if self.ram < 8000:
+			usable_ram = self.ram - 2000  # in MB (leaving some for disk cache + others)
+		elif self.ram < 4000:
+			usable_ram = self.ram - 1500  # in MB (leaving some for disk cache + others)
+		else:
+			usable_ram = max(
+				self.ram - 3000, self.ram * 0.75
+			)  # in MB (leaving some for disk cache + others)
+		if self.is_self_hosted and (self.name == self.database_server):
+			usable_ram -= 2000  # More ram for DB
```

Comment:
This is probably not enough for large DBs. 

This should be roughly equal to bufferpool size that you intend to set. 

If DB starts swapping it will be very bad for performance. 


Is there no way to hard partition memory and "reserve" (not limit) portion of it for a process?

Reply:
Setting to Bufferpool size is doable. But I don't think we can hard partition for a process.